### PR TITLE
ARTEMIS-1169 - Implement Interceptors for the AMQP protocol

### PR DESCRIPTION
**Pull request by andytaylor**
https://issues.apache.org/jira/browse/ARTEMIS-1169

### CODE REVIEW

File: examples/features/standard/interceptor-client-amqp/pom.xml [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            <groupId>org.apache.activemq</groupId>
+            <artifactId>artemis-maven-plugin</artifactId>
+            <executions>
+               <execution>
+                  <id>create</id>
+                  <goals>
+                     <goal>create</goal>
+                  </goals>
+                  <configuration>
+                     <javaOptions>-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=8000</javaOptions>
```
Comment (jbertram) [inline examples/features/standard/interceptor-client-amqp/pom.xml]: This should be removed as it's preventing the example from running.

File: examples/features/standard/interceptor-client-amqp/src/main/java/org/apache/activemq/artemis/amqp/example/InterceptorExample.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+import javax.jms.Connection;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Queue;
+import javax.jms.Session;
+import javax.jms.TextMessage;
+
+/**
+ * A simple example that shows how to implement and use interceptors with ActiveMQ Artemis with the MQTT protocol.
```
Comment (jbertram) [inline examples/features/standard/interceptor-client-amqp/src/main/java/org/apache/activemq/artemis/amqp/example/InterceptorExample.java]: Wrong protocol referenced here. Should be AMQP not MQTT.

Comment (andytaylor) [inline examples/features/standard/interceptor-client-amqp/src/main/java/org/apache/activemq/artemis/amqp/example/InterceptorExample.java]: fixed

**Comment by jbertram** [comment]
The example reports SUCCESS, but it actually fails. It's looking for the property "newSimpleAmqpInterceptor" which should be set by the interceptor, but the interceptor just prints the value of the "SimpleAmqpInterceptor" property.

**Comment by jbertram** [comment]
The PR check passes, but 'mvn -Prelease install' due to checkstyle violations in the new example.

**Comment by andytaylor** [comment]
fixed

**Pull request closed without merging**